tests/kernel/nfs_server.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: NFS server
+#    This module provisions the NFS server and then runs some
+#    basic sanity tests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use Mojo::Base "opensusebasetest";
+use testapi;
+use serial_terminal "select_serial_terminal";
+use lockapi;
+use utils;
+use Utils::Logging "export_logs_basic";
+
+sub run {
+    select_serial_terminal();
+
+    #TODO: configure nfs config as pleased, specifically:
+    #USE_KERNEL_NFSD_NUMBER
+    #NFS3_SERVER_SUPPORT
+    #NFS4_SUPPORT
+    #NFSV4LEASETIME
+    #TODO: introduce nfs ha setup
+
+    #TODO: all below folder creation and exports should be abstracted
+    #TODO: allow this to be configurable
+    my $nfs_mount_nfs3 = "/nfs/shared_nfs3";
+    my $nfs_mount_nfs4 = "/nfs/shared_nfs4";
+    my $nfs_permissions = "rw,sync,no_root_squash";
+
+    my $nfs_mount_nfs3_async = "/nfs/shared_nfs3_async";
+    my $nfs_mount_nfs4_async = "/nfs/shared_nfs4_async";
+    my $nfs_permissions_async = "rw,async,no_root_squash";
+
+    # provision NFS server(s) of various types
+    zypper_call("in yast2-nfs-server");
```

Comment:
maybe we need a dependency list here or somewhere to install the following packages: nfs-kernel-server, nfs-client, nfs4-acl-tools etc...

Reply:
`nfs-kernel-server` and `nfs-client` are mandatory. `yast2-nfs-server` only [*Recommends* `nfs-kernel-server`](https://build.opensuse.org/package/view_file/openSUSE:Factory/yast2-nfs-server/yast2-nfs-server.spec?expand=1). It does not harm to install `nfs4-acl-tools`, although not sure if any of `nfs4_getfacl`, `nfs4_setfacl`, `nfs4_editfacl` are going to be used.